### PR TITLE
Rearrange feature sections

### PR DESCRIPTION
**Pull request by trishalim**
Divide features section into 3 sections: local-first, collaboration, and the rest
Remove redundant sections
Move richtext to "coming soon" section

### CODE REVIEW

**Comment by aeplay** [comment]
@trishalim before you go to far with the illustrations, I don't think the two-way state binding feature is as good as the others to have there.

Not because it's wrong (it's correct), but because people have a very strong "two-way binding bad" preconception since react, so best to just trick them into it and talk about something else here

Plus it's not really an "additional feature" it's more like an inherent feature of the CoValues.

Will think of something to replace it with!

**Comment by trishalim** [comment]
@aeplay Great. I was just about to ask you

**Comment by trishalim** [comment]
@aeplay what do you think about the new placement for e2e?

**Comment by aeplay** [comment]
> @aeplay what do you think about the new placement for e2e?

good place, but could be more prominent! I think it might even deserve more text since it is one of the most unique aspects of Jazz (happy to help with the copy)

**Comment by aeplay** [comment]
And what do you think about server workers as an "extra feature"? We don't really mention it anywhere except in the diagram and it is typically one of the biggest surprises when explaining Jazz

**Comment by trishalim** [comment]
@aeplay ok. Can you write some copy? Any idea for the illustration?

I will try to make e2e more prominent. And feel free to suggest more words for it.

**Comment by aeplay** [comment]
**Server workers**

Subscribe to Jazz state and update existing databases or third-party APIs. Or expose an HTTP API that mutates Jazz state.

**Comment by aeplay** [comment]
Illustration: how about a variant of the hero illustration? Something like this (see Figma) as a very rough sketch

Screenshot 2024-11-01 at 13 51 46

Edit: maybe a zoomed-in part of this to make it simpler?

**Comment by trishalim** [comment]
@aeplay I pushed a commit for e2e 🤔

**Comment by aeplay** [comment]
The big icon helps!

more words for e2ee:

**End-to-end encrypted and tamper-proof**

Instead of trusting a centralised backend to enforce permissions, Jazz uses public-key cryptography. Your edits are encrypted and signed on-device, verifiable by everyone and readable only by users and server workers explicitly given access. The syncing server never sees your data in plaintext.

I know it's long but thoughts?

**Comment by aeplay** [comment]
Also wdyt about "Taking collaboration a step further" -> "Making secure collaboration the default" ?

**Comment by trishalim** [comment]
I think the collaboration section is mostly about the reader, writer, or admin roles. Why do you want to emphasize "secure"?

**Comment by aeplay** [comment]
Plus one more thought for the local-first section:

Offline first + support for bad connections deserves it's own tile.

Here we can talk about our opinion that even in a well connected world offline support and especially support for weak connections is really important and that local-first apps deal with this much more gracefully

**Comment by trishalim** [comment]
> Offline first + support for bad connections deserves it's own tile.

Yes totally 💯

**Comment by trishalim** [comment]
You wanted to make a bold claim about "local first is the best way to build apps" and maybe it's appropriate here

**Comment by aeplay** [comment]
> I think the collaboration section is mostly about the reader, writer, or admin roles. Why do you want to emphasize "secure"?

ok how about "Collaboration as a first-class concept"

**Comment by aeplay** [comment]
> You wanted to make a bold claim about "local first is the best way to build apps" and maybe it's appropriate here

The whole page is about that... I think we're doing a good job of saying that

**Comment by trishalim** [comment]
image

**Comment by aeplay** [comment]
> image

@trishalim nice simplification, but the arrow on the left needs to be reversed, it's HTTP request coming in

**Comment by trishalim** [comment]
image

**Comment by trishalim** [comment]
I added more text to e2ee and the server worker illustration

**Comment by trishalim** [comment]
It would be nice to have social proof from SA1 related to the collaboration features

**Comment by aeplay** [comment]
@trishalim Lol you weren't kidding with HUGE.

I'd say with it having more text it's prominent enough even with a small icon - you can make it look the same as the features in "Why local-first?" for consistency. Wdyt?

social proof

"We just wanted to build a single-player experience first, planning to add team and org features much later. But because of Jazz, we had that from day one. All we needed to add was an invite button."

- Lead Developer @ Invoice Radar

Edit: made the quote sound more natural

**Comment by trishalim** [comment]
So much space on the right 🤔
image

**Comment by aeplay** [comment]
> So much space on the right 🤔 image

Screenshot 2024-11-01 at 15 52 18

@trishalim another diagram will fix it!
(see Figma, you'll be able to make it nicer)

**Comment by trishalim** [comment]
@aeplay I think we have enough diagrams 🙈 I'll think of some other option

**Comment by aeplay** [comment]
> @aeplay I think we have enough diagrams 🙈 I'll think of some other option

@trishalim something like the "random encrypted letters" as in the https://evervault.com hero?
(without the animation)

**Comment by trishalim** [comment]
Cool! I like it

It'd be great if we can cut down the description by one sentence

**Comment by aeplay** [comment]
> Cool! I like it
> 
> It'd be great if we can cut down the description by one sentence

how about

End-to-end encrypted and tamper-proof
The syncing server never sees your data in plaintext.

Instead of trusting centralised authorisation, Jazz uses public-key cryptography. Your edits are encrypted and signed on-device, verifiable by everyone and readable only to those given access.

**Comment by aeplay** [comment]
@trishalim edited

**Comment by trishalim** [comment]
Updated e2e text and illustration
I'm pretty happy with this

**Comment by trishalim** [comment]
Since we have 4 items here, I am trying out a different layout

image

Idk what goes on the right yet, but I'm thinking we could somehow emphasize that it works offline.

That description under the heading needs some work too

**Comment by trishalim** [comment]
@aeplay I'm keeping the local-first section simple for now. I just checked https://www.instantdb.com/ and realized they have a very similar section, and they show small demos each. Maybe we can do a demo as well.

File: homepage/homepage/components/home/EncryptionSection.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    "VBXWFFIX+WVFRNM+CGT",
+    "J",
+    "HYCSL#QN%IYPMPLQUKUJ",
+    "A",
+    "KBANJTYRQ!OUTYAO",
+    "Z",
+];
+
+function Illustration() {
+    return (
```
Comment (aeplay) [inline homepage/homepage/components/home/EncryptionSection.tsx:33]: I'm assuming the lock icon is somewhere in here...

Please may we use https://lucide.dev/icons/lock-keyhole and make it look like one of the feature-chips higher up? (Light blue bg with blue lines in bright mode)

Right now the solid-filled icon looks a bit out of place.

And then you can also use the same icon in the hero for the E2EE feature hint.

Thank you!

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+            </div>
+        ),
+    },
+    {
+        title: "Progressive image loading",
+        icon: ImageIcon,
+        description: (
+            <>
+                Using Jazz&apos;s <code>ImageDefinition</code> component, you
+                get progressive image loading, super fast blur preview, and
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                get progressive image up & downloading, super fast blur preview, and
```

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+                />
+            </>
+        ),
+    },
+    {
+        title: "Server workers",
+        icon: ImageIcon,
+        description: (
+            <>
+                Subscribe to Jazz state and update existing databases or
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                Expose an HTTP API that mutates Jazz state.
```

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+                third-party APIs. Or expose an HTTP API that mutates Jazz state.
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                Or subscribe to Jazz state and update existing databases or third-party APIs.
```

(Changing the order to correspond how you would read the diagram)

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            </>
+        ),
+        illustration: <ServerWorkersDiagram className="pt-8 w-auto" />,
+    },
+    {
+        title: "Authentication",
+        icon: UserIcon,
+        description: (
+            <>
+                Plug and play different kinds of auth like WebAuthN (Touch ID,
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                Plug and play different kinds of auth like Passkeys (Touch ID,
```

File: homepage/homepage/components/home/EncryptionSection.tsx [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+                    <Prose className="max-w-2xl mt-2 md:mt-4">
+                        <p>
+                            <strong>
+                                The syncing server never sees your data in
+                                plaintext.
+                            </strong>{" "}
+                            Instead of trusting centralised authorisation, Jazz
+                            uses public-key cryptography. Your edits are
+                            encrypted and signed on-device, verifiable by
+                            everyone and readable only to those given access.
```
Comment (aeplay) [inline homepage/homepage/components/home/EncryptionSection.tsx]:
```suggestion
                            everyone and readable only by those given access.
```

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+                <div className="border p-4 sm:p-8 shadow-sm rounded-xl col-span-2 sm:col-span-4 space-y-5">
+                    <H3>Jazz Cloud</H3>
+                    <Prose className="max-w-xl">
+                        <p>
+                            Jazz Cloud is a real-time sync and storage
+                            infrastructure that scales your Jazz app up to
+                            millions of users.{" "}
+                            <strong>
+                                Easy setup, no configuration needed.
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                                Instant setup, no configuration needed.
```

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+                            </strong>
+                        </p>
+                    </Prose>
+                    <ul className="flex flex-col sm:flex-row gap-4 text-sm">
+                        {[
+                            "Blob storage",
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                            "Data & blob storage",
```

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 9]
```diff
@@ -0,0 +1,195 @@
...
+                            "Data storage",
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                            "Global sync",
```

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 10]
```diff
@@ -0,0 +1,195 @@
...
+                                <span className="text-blue p-1 rounded-full bg-blue-50 dark:text-blue-500 dark:bg-white/10">
+                                    <CheckIcon size={12} strokeWidth={3} />
+                                </span>
+                                {feature}
+                            </li>
+                        ))}
+                    </ul>
+                    <div className="flex items-center flex-wrap gap-x-5 flex-wrap gap-y-3">
+                        <Button href="/cloud" variant="primary">
+                            View full pricing
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                            View free tier & pricing
```

File: homepage/homepage/components/home/FeaturesSection.tsx [hunk 11]
```diff
@@ -0,0 +1,195 @@
...
+                        </Button>
+                    </div>
+                    <Prose size="sm">
+                        You can rely on us, or{" "}
```
Comment (aeplay) [inline homepage/homepage/components/home/FeaturesSection.tsx]:
```suggestion
                        Or{" "}
```

File: homepage/homepage/components/home/HeroSection.tsx [hunk 12]
```diff
@@ -1,12 +1,53 @@
...
+        title: "Social features",
+        icon: UsersIcon,
+    },
+    {
+        title: "Permissions",
+        icon: FileLock2Icon,
+    },
+    {
+        title: "E2E encryption",
+        icon: KeyRoundIcon,
```
Comment (aeplay) [inline homepage/homepage/components/home/HeroSection.tsx]: https://lucide.dev/icons/lock-keyhole please

File: homepage/homepage/components/home/LocalFirstFeaturesSection.tsx [hunk 13]
```diff
@@ -0,0 +1,80 @@
...
+                </>
+            ),
+        },
+        {
+            title: "Instant updates",
+            icon: GaugeIcon,
+            description: (
+                <>
+                    Since you&apos;re working with local state, your UI updates
+                    instantly. Just mutate data the JSON object. No API calls
```
Comment (aeplay) [inline homepage/homepage/components/home/LocalFirstFeaturesSection.tsx]:
```suggestion
                    instantly. Just mutate data. No API calls
```

File: homepage/homepage/components/home/LocalFirstFeaturesSection.tsx [hunk 14]
```diff
@@ -0,0 +1,80 @@
...
+                    everything in sync.
+                </>
+            ),
+        },
+        {
+            title: "Multiplayer",
+            icon: MousePointerSquareDashedIcon,
+            description: (
+                <>
+                    Adding multiplayer is as easy as sharing state with other
```
Comment (aeplay) [inline homepage/homepage/components/home/LocalFirstFeaturesSection.tsx]:
```suggestion
                    Adding multiplayer is as easy as sharing synced data with other
```

**Review by aeplay — CHANGES_REQUESTED** [review]
Great work!!

A bunch of small suggestions:

**Comment by trishalim** [comment]
@aeplay Thanks for the feedback. Can you please check again

**Review by aeplay — CHANGES_REQUESTED** [review]
Final super annoying request: please make the lock icon on top of the random letters as small as the icons in "why local-first"

Thank you!

**Comment by trishalim** [comment]
No worries. It's small now

File: homepage/homepage/app/(home)/page.tsx [hunk 15]
```diff
@@ -1,217 +1,25 @@
...
+            <HeroSection />
 
-            <BeforeAfterJazz />
+            <ProblemStatementSection />
 
             <div className="container flex flex-col gap-12 mt-12 lg:gap-20 lg:mt-20">
-                <HowItWorks />
+                <HowJazzWorksSection />
 
                 <Testimonial name="Serious Adopter #4" role="Technical Founder">
```
Comment (aeplay) [inline homepage/homepage/app/(home)/page.tsx]:
```suggestion
                <Testimonial name="Spreadsheet app (stealth)" role="CTO">
```

File: homepage/homepage/app/(home)/page.tsx [hunk 16]
```diff
@@ -224,265 +32,30 @@ export default function Home() {
...
-                    <GridCard>
-                        <H3>Auto-sub</H3>
-                        <P className="text-lg">
-                            Let your UI drive data-syncing.
-                        </P>
-                        <Prose size="sm">
-                            <AutoSubDescription />
-                        </Prose>
-                    </GridCard>
+                <Testimonial name="Serious Adopter #1" role="Lead Developer">
```
Comment (aeplay) [inline homepage/homepage/app/(home)/page.tsx]:
```suggestion
                <Testimonial name="Invoice Radar" role="Technical Founder">
```

**Review by aeplay — APPROVED** [review]

File: homepage/homepage/components/home/CollaborationPrivate.mdx [hunk 17]
```diff
@@ -0,0 +1,8 @@
+```tsx
+const { me } = useAccount();
+
+Message.create({
```
Comment (aeplay) [inline homepage/homepage/components/home/CollaborationPrivate.mdx]:
```suggestion
Message.create(
```

File: homepage/homepage/components/home/CollaborationPrivate.mdx [hunk 18]
```diff
@@ -0,0 +1,8 @@
...
+  text: ...,
```
Comment (aeplay) [inline homepage/homepage/components/home/CollaborationPrivate.mdx]:
```suggestion
  { text: ... },
```

File: homepage/homepage/components/home/CollaborationPrivate.mdx [hunk 19]
```diff
@@ -0,0 +1,8 @@
...
+  owner: me
```
Comment (aeplay) [inline homepage/homepage/components/home/CollaborationPrivate.mdx]:
```suggestion
  { owner: me }
```

File: homepage/homepage/components/home/CollaborationPrivate.mdx [hunk 20]
```diff
@@ -0,0 +1,8 @@
...
+});
```
Comment (aeplay) [inline homepage/homepage/components/home/CollaborationPrivate.mdx]:
```suggestion
);
```

File: homepage/homepage/components/home/CollaborationPublic.mdx [hunk 21]
```diff
@@ -0,0 +1,7 @@
+```tsx
+const group = Group.create({
+  owner: me
+});
+
+group.addMember("everyone", "writer");
```
Comment (aeplay) [inline homepage/homepage/components/home/CollaborationPublic.mdx:6]:
```suggestion
group.addMember("everyone", "writer");

Chat.create([], { owner: group })
```

**Pull request merged**